unstable/lens/hash-slice.rkt

```diff
@@ -0,0 +1,26 @@
+#lang racket/base
+
+(provide hash-slice-lens)
+
+(require lens/base/main)
+(module+ test
+  (require rackunit))
+
+(define (hash-slice-lens . ks)
+  (define (get hsh)
+    (for/hash ([k (in-list ks)])
+      (values k (hash-ref hsh k))))
+  (define (set hsh hsh2)
+    (define hsh1
+      (for/fold ([hsh hsh]) ([k (in-list ks)])
+        (hash-remove hsh k)))
+    (for/fold ([hsh hsh1]) ([(k v) (in-hash hsh2)])
+      (hash-set hsh k v)))
+  (make-lens get set))
+
+(module+ test
+  (check-equal? (lens-view (hash-slice-lens 'a 'c) (hash 'a 1 'b 2 'c 3))
+                (hash 'a 1 'c 3))
+  (check-equal? (lens-set (hash-slice-lens 'a 'c) (hash 'a 1 'b 2 'c 3) (hash 'd 4))
```

Comment:
If you `set` using keys that aren't in the view, you can't see the new values for those keys because the lens doesn't view them. The lens manipulates things outside it's view. Specifically it breaks set-get consistency.

```racket
> (define a-lens (hash-slice-lens 'a))
> (lens-set (hash 'a 4) (hash 'a 4 'b 2))
(hash 'a 4 'b 2)
> (lens-view (hash 'a 4 'b 2))
(hash 'a 4) ; should have viewed exactly what was set above, (hash 'a 4 'b 2)
```